avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java

```diff
@@ -76,4 +68,7 @@
       throw new SerializationException("Error serializing Avro message", e);
     }
   }
+
+  protected abstract DatumWriter<Object> getDatumWriter(Schema schema, Object object);
```

Comment:
Good question, I'm definitely not happy about having that logic everywhere. Reason is, as these classes are currently structured, the "leaf" classes deriving from `AbstractKafkaAvroSerializer` (`AvroMessageReader`, `KafkaAvroSerializer` and `KafkaAvroDecoder`) all do the config parsing themselves, and maintain any related objects. `getDatumWriter` needs to access the configured `DatumWriter` implementation, which only the leaf classes know about. 

One option I can think of:

- Have an abstract `getDatumWriterClass` instead, so instead of each subclass knowing how to instantiate the correct `DatumWriter`, they just need to be able to tell superclass which class to use.

This might be easier to improve once #162 is merged as config is reified to separate object.